### ISSUE TITLE
colo instance lost storage

### ISSUE THREAD

**Issue by citrus-it** [oxidecomputer/omicron#4775]
Instance ID d54adb97-2508-4ceb-92f0-4d7e0c650893 on rack3/colo lost access to its boot disk over the weekend, panicked and was unable to reboot. I'm opening this issue here for triage and it may need to be moved to a different repo or split up later.

This instance has two disks:

```
pptang% oxide instance disk list --instance d54adb97-2508-4ceb-92f0-4d7e0c650893
Disk {
    block_size: ByteCount(
        512,
    ),
    description: "",
    device_path: "/mnt/ox-us-west-boot",
    id: bee14a79-296f-46e9-af09-1f9b8ccb1e60,
    image_id: Some(
        6f10ec23-e6a0-48eb-abe0-75939de4a790,
    ),
    name: Name(
        "ox-us-west-boot",
    ),
    project_id: ae90ba27-b4f1-4341-a4c7-4cababe1b3a0,
    size: ByteCount(
        85899345920,
    ),
    snapshot_id: None,
    state: Attached(
        d54adb97-2508-4ceb-92f0-4d7e0c650893,
    ),
    time_created: 2023-12-15T16:58:37.981728Z,
    time_modified: 2023-12-15T16:58:37.981728Z,
}
Disk {
    block_size: ByteCount(
        4096,
    ),
    description: "",
    device_path: "/mnt/ox-us-west-data",
    id: b4ad7d23-508a-4994-b79e-46e62bda9273,
    image_id: None,
    name: Name(
        "ox-us-west-data",
    ),
    project_id: ae90ba27-b4f1-4341-a4c7-4cababe1b3a0,
    size: ByteCount(
        322122547200,
    ),
    snapshot_id: None,
    state: Attached(
        d54adb97-2508-4ceb-92f0-4d7e0c650893,
    ),
    time_created: 2023-12-15T16:58:55.815127Z,
    time_modified: 2023-12-15T16:58:55.815127Z,
}
```

and it appears that disk `bee14a79-296f-46e9-af09-1f9b8ccb1e60` suddenly went away from the guest's perspective. The propolis-server for the instance contains these lines:

```
{"msg":"[0] downstairs disconnected","v":0,"name":"propolis-server","level":40,"time":"2024-01-05T19:36:56.331298805Z","hostname":"oxz_propolis-server_32e75239-9fe7-437f-8ad1-890c146b352a","pid":14610,"session_id":"2e989329-97c5-41df-bc17-f6050b42ea04","component":"crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60"}
{"msg":"[1] downstairs disconnected","v":0,"name":"propolis-server","level":40,"time":"2024-01-05T19:36:56.35713016Z","hostname":"oxz_propolis-server_32e75239-9fe7-437f-8ad1-890c146b352a","pid":14610,"session_id":"2e989329-97c5-41df-bc17-f6050b42ea04","component":"crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60"}
{"msg":"[2] downstairs disconnected","v":0,"name":"propolis-server","level":40,"time":"2024-01-05T19:36:56.426927816Z","hostname":"oxz_propolis-server_32e75239-9fe7-437f-8ad1-890c146b352a","pid":14610,"session_id":"2e989329-97c5-41df-bc17-f6050b42ea04","component":"crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60"}
```

and is now repeatedly logging:

```
11:16:26.488Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] connecting to [fd00:1122:3344:101::9]:19002
11:16:26.488Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] connecting to [fd00:1122:3344:11f::4]:19001
11:16:26.488Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] connecting to [fd00:1122:3344:113::9]:19001
```

These connection attempts are all getting TCP connection refused from the destination.

Let's have a look at this disk:

```
root@oxz_switch:~# /tmp/omdb db disks info bee14a79-296f-46e9-af09-1f9b8ccb1e60
HOST_SERIAL DISK_NAME       INSTANCE_NAME PROPOLIS_ZONE                                            VOLUME_ID                            DISK_STATE
BRM44220012 ox-us-west-boot ox-us-west    oxz_propolis-server_32e75239-9fe7-437f-8ad1-890c146b352a 89b99584-6b75-4872-9c20-f64b95ee94ad attached
HOST_SERIAL REGION                               ZONE                                              PHYSICAL_DISK
BRM42220042 4cc4c7d2-b842-422b-8086-029826cf0641 oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7 e9f13dca-df4e-4663-ac3b-69caf1ef1ccf
BRM42220053 4dee5ce1-a2d3-44f5-acc2-fd1d84f59ac1 oxz_crucible_5100e222-5ea4-4e67-9040-679137e666c8 1db9258f-b4f1-4262-98ff-ae2228e0eddd
BRM42220035 f8611743-6a7f-4334-aa59-b739a241b5e9 oxz_crucible_d802baae-9c3f-437a-85fe-cd72653b6db1 97140495-bdc3-4c19-94f7-ff1d7af6f7f3
```

and picking the first downstairs in that list:

```

root@oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7:~# ipadm
ADDROBJ           TYPE     STATE        ADDR
lo0/v4            static   ok           127.0.0.1/8
lo0/v6            static   ok           ::1/128
oxControlService9/ll addrconf ok        fe80::8:20ff:fea4:dfc9%oxControlService9/10
oxControlService9/omicron6 static ok    fd00:1122:3344:119::b/64
```

Well, that seems odd. The IP address is not one of the ones that crucible in propolis-server is trying to connect to. Let's summarise:

| crucible zone | IP address |
| ---- | ---- |
| oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7 | fd00:1122:3344:119::b |
| oxz_crucible_5100e222-5ea4-4e67-9040-679137e666c8  | fd00:1122:3344:11e::5 |
| oxz_crucible_d802baae-9c3f-437a-85fe-cd72653b6db1 | fd00:1122:3344:108::5 |

None of these correspond to the addresses that propolis-server is trying to connect to.

The crucible zone that has `fd00:1122:3344:101::9` (the first address propolis is trying), is `oxz_crucible_6a0baf13-a80b-4778-a0ab-a69cd851de2d` on `BRM42220048`, and is not running a downstairs on port 19002.

```
root@oxz_crucible_6a0baf13-a80b-4778-a0ab-a69cd851de2d:~# ipadm
ADDROBJ           TYPE     STATE        ADDR
oxControlService11/omicron6 static ok   fd00:1122:3344:101::9/64

root@oxz_crucible_6a0baf13-a80b-4778-a0ab-a69cd851de2d:~# netstat -an | grep LISTEN
fd00:1122:3344:101::9.19003             *.*                              0      0 128000      0 LISTEN
fd00:1122:3344:101::9.23003             *.*                              0      0 128000      0 LISTEN
fd00:1122:3344:101::9.19000             *.*                              0      0 128000      0 LISTEN
fd00:1122:3344:101::9.23000             *.*                              0      0 128000      0 LISTEN
```

Going back to `oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7` - the first region for this disk according to the database, there is nothing logged for region `4cc4c7d2-b842-422b-8086-029826cf0641` after the original instance boot:

```
...
{"msg":"UpstairsConnection { upstairs_id: bee14a79-296f-46e9-af09-1f9b8ccb1e60, session_id: 3c3dd83f-4808-4f44-8611-e7e355a3a8d2, gen: 1 } is now active (read-write)","v":0,"name":"crucible","level":30,"time":"2023-12-15T16:59:33.571399142Z","hostname":"oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7","pid":14744}
{"msg":"Current flush_numbers [0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]","v":0,"name":"crucible","level":30,"time":"2023-12-15T16:59:33.579531685Z","hostname":"oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7","pid":14744,"task":"proc"}
{"msg":"Downstairs has completed Negotiation","v":0,"name":"crucible","level":30,"time":"2023-12-15T16:59:33.579673915Z","hostname":"oxz_crucible_197695e1-d949-4982-b679-6e5c9ab4bcc7","pid":14744,"task":"proc"}
```

Going back to the instance, the logs show the original crucible startup:

```
16:59:33.512Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Upstairs starts
16:59:33.512Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Crucible Version: BuildInfo {
        version: "0.0.1",
        git_sha: "fab27994d0bd12725c17d6b478a9bfc2673ad6f4",
        git_commit_timestamp: "2023-12-08T22:28:34.000000000Z",
        git_branch: "master",
        rustc_semver: "1.74.1",
        rustc_channel: "stable",
        rustc_host_triple: "x86_64-unknown-illumos",
        rustc_commit_sha: "a28077b28a02b92985b3a3faecf92813155f1ea1",
        cargo_triple: "x86_64-unknown-illumos",
        debug: false,
        opt_level: 3,
    }
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Upstairs <-> Downstairs Message Version: 5
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Proc runs for [fd00:1122:3344:11e::5]:19000 in state New
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Using region definition RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 128, uuid: 00000000-0000-0000-0000-000000000000, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Crucible 61ab496d-f467-41cc-9ecf-a67d9ec8e32b has session id: 2e989329-97c5-41df-bc17-f6050b42ea04
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Upstairs opts: Upstairs UUID: 61ab496d-f467-41cc-9ecf-a67d9ec8e32b, Targets: [[fd00:1122:3344:113::9]:19001, [fd00:1122:3344:11f::4]:19001, [fd00:1122:3344:101::9]:19002], lossy: false, flush_timeout: None, key populated: true,  cert_pem populated: false,  key_pem populated: false,  root_cert_pem populated: false,  Control: None,  read_only: true
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Crucible stats registered with UUID: 61ab496d-f467-41cc-9ecf-a67d9ec8e32b
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] bee14a79-296f-46e9-af09-1f9b8ccb1e60 looper connected
    looper = 2
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Proc runs for [fd00:1122:3344:119::b]:19001 in state New
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] connecting to [fd00:1122:3344:113::9]:19001
    looper = 0
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): up_listen starts
    task = up_listen
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Wait for all three downstairs to come online
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Flush timeout: 0.5
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] connecting to [fd00:1122:3344:101::9]:19002
    looper = 2
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] connecting to [fd00:1122:3344:11f::4]:19001
    looper = 1
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub check for bee14a79-296f-46e9-af09-1f9b8ccb1e60
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub pause 120 seconds before starting
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b looper connected
    looper = 2
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Proc runs for [fd00:1122:3344:101::9]:19002 in state New
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b looper connected
    looper = 0
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Proc runs for [fd00:1122:3344:113::9]:19001 in state New
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b looper connected
    looper = 1
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Proc runs for [fd00:1122:3344:11f::4]:19001 in state New
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) New New New ds_transition to WaitActive
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Transition from New to WaitActive
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) New New WaitActive ds_transition to WaitActive
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Transition from New to WaitActive
16:59:33.513Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) New New New ds_transition to WaitActive
16:59:33.514Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Transition from New to WaitActive
16:59:33.537Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) WaitActive New WaitActive ds_transition to WaitActive
16:59:33.537Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Transition from New to WaitActive
16:59:33.542Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) New New WaitActive ds_transition to WaitActive
16:59:33.542Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Transition from New to WaitActive
16:59:33.544Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) WaitActive New WaitActive ds_transition to WaitActive
16:59:33.544Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Transition from New to WaitActive
16:59:33.571Z INFO propolis-server (vm_controller): Sending startup complete to block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60
16:59:33.571Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 active request set
16:59:33.571Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] received activate with gen 1
16:59:33.571Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] client got ds_active_rx, promote! session 3c3dd83f-4808-4f44-8611-e7e355a3a8d2
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] received activate with gen 1
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] client got ds_active_rx, promote! session 3c3dd83f-4808-4f44-8611-e7e355a3a8d2
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] received activate with gen 1
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] client got ds_active_rx, promote! session 3c3dd83f-4808-4f44-8611-e7e355a3a8d2
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] downstairs client at [fd00:1122:3344:119::b]:19001 has UUID 4cc4c7d2-b842-422b-8086-029826cf0641
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 1280, uuid: 4cc4c7d2-b842-422b-8086-029826cf0641, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitActive WaitActive WaitActive
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] downstairs client at [fd00:1122:3344:11e::5]:19000 has UUID 4dee5ce1-a2d3-44f5-acc2-fd1d84f59ac1
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 1280, uuid: 4dee5ce1-a2d3-44f5-acc2-fd1d84f59ac1, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.572Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitActive WaitActive WaitActive
16:59:33.577Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] downstairs client at [fd00:1122:3344:108::5]:19002 has UUID f8611743-6a7f-4334-aa59-b739a241b5e9
16:59:33.577Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 1280, uuid: f8611743-6a7f-4334-aa59-b739a241b5e9, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.577Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitActive WaitActive WaitActive
16:59:33.578Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) WaitActive WaitActive WaitActive ds_transition to WaitQuorum
16:59:33.578Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Transition from WaitActive to WaitQuorum
16:59:33.578Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] new RM replaced this: None
16:59:33.578Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Starts reconcile loop
16:59:33.578Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] [fd00:1122:3344:11e::5]:19000 task reports connection:true
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitQuorum WaitActive WaitActive
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Waiting for 2 more clients to be ready
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) WaitQuorum WaitActive WaitActive ds_transition to WaitQuorum
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Transition from WaitActive to WaitQuorum
16:59:33.579Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] new RM replaced this: None
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Starts reconcile loop
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] [fd00:1122:3344:119::b]:19001 task reports connection:true
16:59:33.579Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitQuorum WaitActive WaitQuorum
16:59:33.580Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Waiting for 1 more clients to be ready
16:59:33.580Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.588Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] bee14a79-296f-46e9-af09-1f9b8ccb1e60 (3c3dd83f-4808-4f44-8611-e7e355a3a8d2) WaitQuorum WaitActive WaitQuorum ds_transition to WaitQuorum
16:59:33.588Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Transition from WaitActive to WaitQuorum
16:59:33.588Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] new RM replaced this: None
16:59:33.588Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] [fd00:1122:3344:108::5]:19002 task reports connection:true
16:59:33.588Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 WaitQuorum WaitQuorum WaitQuorum
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R flush_numbers[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R generation[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R flush_numbers[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R generation[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R flush_numbers[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R generation[0..12]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Max found gen is 1
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Generation requested: 1 >= found:1
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Next flush: 1
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): All extents match
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): No downstairs repair required
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): No initial repair work was required
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set Downstairs and Upstairs active
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 is now active with session: 3c3dd83f-4808-4f44-8611-e7e355a3a8d2
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): bee14a79-296f-46e9-af09-1f9b8ccb1e60 Set Active after no repair
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Notify all downstairs, region set compare is done.
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] received reconcile message
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] All repairs completed, exit
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Starts cmd_loop
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] received reconcile message
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] All repairs completed, exit
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] received reconcile message
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] All repairs completed, exit
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Starts cmd_loop
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Starts cmd_loop
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b active request set
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] received activate with gen 13
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] client got ds_active_rx, promote! session 2e989329-97c5-41df-bc17-f6050b42ea04
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] received activate with gen 13
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] client got ds_active_rx, promote! session 2e989329-97c5-41df-bc17-f6050b42ea04
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] received activate with gen 13
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] client got ds_active_rx, promote! session 2e989329-97c5-41df-bc17-f6050b42ea04
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] downstairs client at [fd00:1122:3344:101::9]:19002 has UUID a2442c68-2de6-4db0-b07e-59f097fca382
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 128, uuid: a2442c68-2de6-4db0-b07e-59f097fca382, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.590Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitActive WaitActive WaitActive
16:59:33.601Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] downstairs client at [fd00:1122:3344:113::9]:19001 has UUID 5be2f529-faa1-40cc-afcc-223b4e1a8d54
16:59:33.601Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 128, uuid: 5be2f529-faa1-40cc-afcc-223b4e1a8d54, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.601Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitActive WaitActive WaitActive
16:59:33.601Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] downstairs client at [fd00:1122:3344:11f::4]:19001 has UUID 9bc8b7e0-d072-42a1-8fdc-83f83a809400
16:59:33.601Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Got region def RegionDefinition { block_size: 512, extent_size: Block { value: 131072, shift: 9 }, extent_count: 128, uuid: 9bc8b7e0-d072-42a1-8fdc-83f83a809400, encrypted: true, database_read_version: 1, database_write_version: 1 }
16:59:33.602Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitActive WaitActive WaitActive
16:59:33.649Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) WaitActive WaitActive WaitActive ds_transition to WaitQuorum
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Transition from WaitActive to WaitQuorum
16:59:33.650Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] new RM replaced this: None
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Starts reconcile loop
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] [fd00:1122:3344:101::9]:19002 task reports connection:true
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitActive WaitActive WaitQuorum
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Waiting for 2 more clients to be ready
16:59:33.650Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) WaitActive WaitActive WaitQuorum ds_transition to WaitQuorum
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Transition from WaitActive to WaitQuorum
16:59:33.660Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] new RM replaced this: None
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Starts reconcile loop
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] [fd00:1122:3344:11f::4]:19001 task reports connection:true
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitActive WaitQuorum WaitQuorum
16:59:33.660Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Waiting for 1 more clients to be ready
16:59:33.661Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] 61ab496d-f467-41cc-9ecf-a67d9ec8e32b (2e989329-97c5-41df-bc17-f6050b42ea04) WaitActive WaitQuorum WaitQuorum ds_transition to WaitQuorum
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Transition from WaitActive to WaitQuorum
16:59:33.668Z WARN propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] new RM replaced this: None
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Starts reconcile loop
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] [fd00:1122:3344:113::9]:19001 task reports connection:true
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b WaitQuorum WaitQuorum WaitQuorum
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R flush_numbers[0..12]: [2, 0, 0, 0, 100, 190, 289, 390, 0, 0, 0, 0]
16:59:33.668Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R generation[0..12]: [1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R flush_numbers[0..12]: [2, 0, 0, 0, 100, 190, 289, 390, 0, 0, 0, 0]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R generation[0..12]: [1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R flush_numbers[0..12]: [2, 0, 0, 0, 100, 190, 289, 390, 0, 0, 0, 0]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R generation[0..12]: [1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2]R dirty[0..12]: [false, false, false, false, false, false, false, false, false, false, false, false]
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Max found gen is 2
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Generation requested: 13 >= found:2
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Next flush: 1467
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): All extents match
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): No downstairs repair required
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): No initial repair work was required
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set Downstairs and Upstairs active
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b is now active with session: 2e989329-97c5-41df-bc17-f6050b42ea04
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): 61ab496d-f467-41cc-9ecf-a67d9ec8e32b Set Active after no repair
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Notify all downstairs, region set compare is done.
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Set check for repair
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] received reconcile message
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] All repairs completed, exit
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [2] Starts cmd_loop
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] received reconcile message
16:59:33.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] All repairs completed, exit
16:59:33.670Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] received reconcile message
16:59:33.670Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] All repairs completed, exit
16:59:33.670Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [1] Starts cmd_loop
16:59:33.670Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): [0] Starts cmd_loop
16:59:34.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Checking if live repair is needed
16:59:34.589Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): No Live Repair required at this time
16:59:34.669Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Checking if live repair is needed
```

**Comment by citrus-it** [comment]
After further digging, it appears that the guest losing its disk corresponded with the deletion of the image from which the VM was created.

| Timestamp | Note |
| ---- | ---- |
| 2023-12-15 16:58:37.981728+00 | Disk created |
| 2023-12-15 16:59:33.448 | VM booted |
| 2024-01-05 19:36:56.139078+00 | Image deleted |
| 2024-01-05T19:36:56.331298805 | Downstairs disconnected |
| 2024-01-05T19:36:56.35713016 | Downstairs disconnected |
| 2024-01-05T19:36:56.426927816Z | Downstairs disconnected |

**Comment by citrus-it** [comment]
There was a successful scrub pretty soon after the VM boot that culminated in "Submitted removal for read only parent".

```
2023-12-15
17:01:33.519Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub for bee14a79-296f-46e9-af09-1f9b8ccb1e60 begins
17:01:33.519Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub with total_size:8589934592 block_size:512
17:01:33.519Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrubs from block 0 to 16777216 in (256) 131072 size IOs pm:25
17:02:52.059Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 671232/16777216 sp:671232
17:04:11.228Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 1342208/16777216 sp:1342208
17:05:29.447Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 2013440/16777216 sp:2013440
17:06:47.670Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 2684416/16777216 sp:2684416
17:08:06.884Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 3355648/16777216 sp:3355648
17:09:24.971Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 4026624/16777216 sp:4026624
17:10:43.112Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 4697856/16777216 sp:4697856
17:12:02.882Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 5368832/16777216 sp:5368832
17:13:21.102Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 6039808/16777216 sp:6039808
17:14:39.222Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 6711040/16777216 sp:6711040
17:15:58.506Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 7382016/16777216 sp:7382016
17:17:16.792Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 8053248/16777216 sp:8053248
17:18:34.976Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 8724224/16777216 sp:8724224
17:19:53.240Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 9395456/16777216 sp:9395456
17:21:12.224Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 10066432/16777216 sp:10066432
17:22:30.446Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 10737664/16777216 sp:10737664
17:23:48.629Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 11408640/16777216 sp:11408640
17:25:06.762Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 12079616/16777216 sp:12079616
17:26:24.831Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 12750848/16777216 sp:12750848
17:27:43.015Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 13421824/16777216 sp:13421824
17:29:01.188Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 14093056/16777216 sp:14093056
17:30:19.473Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 14764032/16777216 sp:14764032
17:31:37.739Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 15435264/16777216 sp:15435264
17:32:56.635Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 16106240/16777216 sp:16106240
17:34:14.427Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub at offset 16777216/16777216 sp:16777216
17:34:14.452Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub bee14a79-296f-46e9-af09-1f9b8ccb1e60 done in 1960 seconds. Retries:0 scrub_size:131072 size:16777216 pause_milli:25
17:34:14.453Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Scrub of volume bee14a79-296f-46e9-af09-1f9b8ccb1e60 completed, remove parent
17:34:14.664Z INFO propolis-server (crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60): Submitted removal for read only parent on bee14a79-296f-46e9-af09-1f9b8ccb1e60
17:43:40.253Z INFO propolis-server (vm_controller): Sending pause request to block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60
17:43:40.254Z INFO propolis-server (vm_controller): Got paused future from entity block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60
17:43:40.254Z INFO propolis-server (vm_controller): entity block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60 completed pause
17:43:40.254Z INFO propolis-server (vm_controller): Sending reset request to block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60
17:43:40.254Z INFO propolis-server (vm_controller): Sending resume request to block-crucible-bee14a79-296f-46e9-af09-1f9b8ccb1e60
```

**Comment by leftwo** [comment]
After a quick meet, a questions I'm going to dig into, while I attempt to reproduce this myself is:
The guest OS has panic'd after not getting an answer from an IO.  What IO is this?
We don't think that the ROP should be getting any IO requests any longer as the scrub has
finished, but this does not appear to be true.

**Comment by leftwo** [comment]
I did finally get it to happen on my bench gimlet.  I've found a bug in the scrub check code:
https://github.com/oxidecomputer/crucible/issues/1092
And, that's probably what we see here.

There is another issue that came out of this as well: https://github.com/oxidecomputer/crucible/issues/1090
That one you did not hit, but could.

**Comment by leftwo** [comment]
The fix for this was made in crucible https://github.com/oxidecomputer/crucible/commit/e71b10d2f9f1fb52818b916bae83ba15a339548d

And that was rolled up into Omicron at: https://github.com/oxidecomputer/omicron/commit/19059b1bedc8bbc7a9d293486842a9a4fd264ea3

So, we should be all set!
